test/e2e-tests/tests/pipelines_test.go

```diff
@@ -0,0 +1,108 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/opendatahub-io/ai-edge/test/e2e-tests/support"
+	"golang.org/x/net/context"
+	k8sApi "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PipelineRelativePath    = "../../../pipelines/tekton/aiedge-e2e"
+	PipelineRunRelativePath = "../../../pipelines/tekton/aiedge-e2e/aiedge-e2e.pipelinerun.yaml"
+	S3BucketNameEnvKey      = "S3_BUCKET"
+	RegistryRepoNameEnvKey  = "REGISTRY_REPO"
+)
+
+func CreateContext() context.Context {
+	return context.Background()
+}
+
+func WaitForAllPipelineRunsToComplete(ctx context.Context, clients *support.Clients) error {
+	checkPipelineRuns := true
+
+	for checkPipelineRuns {
+		time.Sleep(time.Second * 5)
```

Comment:
Here I meant it a little bit different. The wait loop for conditions to be present was OK, the loop just could wait until the `len(pipelineRun.Status.Conditions) == 0` is false.



So to sum up I was thinking of something like:

```go
list, err := clients.PipelineRun.List(ctx, k8sApi.ListOptions{})

For each pipeline from the list {
    Wait for the conditions to appear in a for loop{}
    pipelineRunning := true
    For pipelineRunning {
        For each condition {
            If Failed, return error, elseif Running wait 5 seconds, else pipelineRunning = false // or similar
        }
    }
}
```
I believe this would work the same. We can wait on a specific pipeline because we anyway want all of them to complete. So we don't need to check all of them periodically, we just need to finish when the last pipeline finishes. Unless the pipelines can be started while this check is already running, in that case you may want to fetch "fresh" pipeline list every time. But here the usual use case will be run all pipelines and then wait.

Reply:
I am kind of confused, but maybe I am just not understanding. In the example above are we checking the same conditions each time, and if so what would make that condition change for us to get out of the `pipelineRunning` loop. I thought we would need to call `PipelineRun.List` to get the up to date info on the condition of each pipelineRun.

Reply:
@jackdelahunt Ah, sorry, I didn't consider that, my bad. However, after fetching the initial pipeline list you can always fetch a single fresh pipeline run before each condition check. This way the loop would be probably cleaner. Also you can merge waiting for the condition to appear loop with the pipelineRunning loop. E.g.:

```go
list, err := clients.PipelineRun.List(ctx, k8sApi.ListOptions{})

For each pipeline from the list {
    pipelineRunning := true
    For pipelineRunning {
        // Here fetch a single pipeline run and get its conditions
        // If no condition present, wait 5 seconds (the next loop will be noop) and let the pipelineRunning loop execute again or you can explicitly use continue
        For each condition {
            If Failed, return error, elseif Running wait 5 seconds, else pipelineRunning = false // or similar
        }
    }
}
```

I think this is cleaner than what was before but this is just my suggestion.

Reply:
I did as you said and I am not really sure if it ended up looking cleaner but you can take a look

Reply:
Looks good to me, however, [this one](https://github.com/jackdelahunt/ai-edge/blob/08b7ffa743ecdd68a161fecdaa65105bccc2df36/test/e2e-tests/tests/pipelines_test.go#L23-L50) looked good as well. The new one won't fetch pipeline runs which are already completed every time the loop is waiting for a specific pipeline to finish. I think there is no nicer way of doing it.